skimage/util/montage.py

```diff
@@ -62,26 +66,36 @@ def montage2d(arr_in, fill='mean', rescale_intensity=False, grid_shape=None):
     >>> arr_out.shape
     (4, 4)
     >>> arr_out
-    array([[  0. ,   1. ,   4. ,   5. ],
-           [  2. ,   3. ,   6. ,   7. ],
-           [  8. ,   9. ,   5.5,   5.5],
-           [ 10. ,  11. ,   5.5,   5.5]])
+    array([[ 0,  1,  4,  5],
+           [ 2,  3,  6,  7],
+           [ 8,  9,  5,  5],
+           [10, 11,  5,  5]])
     >>> arr_in.mean()
     5.5
     >>> arr_out_nonsquare = montage2d(arr_in, grid_shape=(1, 3))
     >>> arr_out_nonsquare
-    array([[  0.,   1.,   4.,   5.,   8.,   9.],
-           [  2.,   3.,   6.,   7.,  10.,  11.]])
+    array([[ 0,  1,  4,  5,  8,  9],
+           [ 2,  3,  6,  7, 10, 11]])
     >>> arr_out_nonsquare.shape
     (2, 6)
 
     """
-
     assert arr_in.ndim == 3
 
-    n_images, height, width = arr_in.shape
+    # -- fill missing patches (needs to be calculated before border padding)
+    if fill == 'mean':
+        fill = arr_in.mean()
 
-    arr_in = arr_in.copy()
+    # -- add border padding, np.pad does all dimensions
+    # so we remove the padding from the first
+    if padding_width > 0:
+        # only pad after to make the width correct
+        bef_aft = (0, padding_width)
+        arr_in = np.pad(arr_in, ((0, 0), bef_aft, bef_aft), mode='constant')
```

Comment:
I sitll think we should pad on all sides of the array (i.e. `bef_aft = (padding_width, padding_width)`)